### PR TITLE
a2600.cpp: Merge tvboy.cpp and a2600.h into a2600.cpp

### PR DESCRIPTION
**Pull request by wilbertpol**

### CODE REVIEW

**Comment by 0kmg** [comment]
Nice, I was thinking of doing this too when I saw that tvboy.cpp was basically a copy paste job of a2600.

One thing I noticed that isn't yet mentioned in the TODO is the fact that most of the TV Boy type units didn't have joystick ports, but they all have D-pad and button controls built into the unit. According to https://www.old-computers.com/museum/computer.asp?st=2&c=827 the builtin controller yields both player inputs at the same time in two player games.

Unrelated but while I'm mentioning discrepancies and TODOs, according to the parts list POP uses a different crystal. 2600 NTSC is part no. C015510, 3.579575 as in the driver. The POP has part no. C016010, at the slightly lower 3.579545.

**Comment by wilbertpol** [comment]
> Nice, I was thinking of doing this too when I saw that tvboy.cpp was basically a copy paste job of a2600.
> 
> One thing I noticed that isn't yet mentioned in the TODO is the fact that most of the TV Boy type units didn't have joystick ports, but they all have D-pad and button controls built into the unit. According to https://www.old-computers.com/museum/computer.asp?st=2&c=827 the builtin controller yields both player inputs at the same time in two player games.
> 
> Unrelated but while I'm mentioning discrepancies and TODOs, according to the parts list POP uses a different crystal. 2600 NTSC is part no. C015510, 3.579575 as in the driver. The POP has part no. C016010, at the slightly lower 3.579545.

Thanks for that information. 
For now I have added a TODO about the controls and made the NTSC and POP unit use different crystals.

File: src/mame/atari/a2600.cpp [hunk 1]
```diff
@@ -87,20 +96,167 @@ E1 Prog ROM 42 DEMON/DIAMOND (CX2615)
...
+#include "machine/mos6530n.h"
+#else
+#include "machine/6532riot.h"
+#endif
+
+//#define VERBOSE (LOG_GENERAL)
+#include "logmacro.h"
 
 
-static constexpr auto MASTER_CLOCK_NTSC = 3.579575_MHz_XTAL;
```
Comment (0kmg) [inline src/mame/atari/a2600.cpp:98]: FYI, there's a commented out line of code on 467 that uses this identifier and is now stale.

Comment (wilbertpol) [reply src/mame/atari/a2600.cpp:98]: Yeah, I know. The commented code even did not compile anymore.
That is now fixed as well

**Pull request merged**